### PR TITLE
image fix lint errors/warnings

### PR DESCRIPTION
**Pull request by brahmaroutu**
Addresses #14756
Signed-off-by: Srini Brahmaroutu srbrahma@us.ibm.com

### CODE REVIEW

File: image/image.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 
 var validHex = regexp.MustCompile(`^([a-f0-9]{64})$`)
 
+// Image - detailed information about image
```
Comment (cpuguy83) [inline image/image.go]: `Images stores the image configuration`

File: image/image.go [hunk 2]
```diff
@@ -26,7 +27,7 @@ type Image struct {
 	Size            int64
 }
 
-// Build an Image object from raw json data
+// NewImgJSON - Build an Image object from raw json data
```
Comment (cpuguy83) [inline image/image.go]: `NewImgJSON creates an Image configuration from json`

File: image/image.go [hunk 3]
```diff
@@ -37,7 +38,7 @@ func NewImgJSON(src []byte) (*Image, error) {
 	return ret, nil
 }
 
-// Check wheather id is a valid image ID or not
+// ValidateID - Check wheather id is a valid image ID or not
```
Comment (cpuguy83) [inline image/image.go]: `ValidateID checks whether an ID string is a valid image ID`

Comment (brahmaroutu) [inline image/image.go]: Added the proper text provided. Thanks @cpuguy83

**Comment by cpuguy83** [comment]
LGTM

**Comment by runcom** [comment]
@brahmaroutu seems like you're committing as `root` and it's not signed could you check pls?

LGTM though

File: image/image.go [hunk 4]
```diff
@@ -11,6 +11,7 @@ import (
 
 var validHex = regexp.MustCompile(`^([a-f0-9]{64})$`)
 
+// Image stores the image configuration
```
Comment (stevvooe) [inline image/image.go]: Again, please document the usage of the type. The goal is not to have `golint` pass.

File: image/image.go [hunk 5]
```diff
@@ -11,6 +11,7 @@ import (
...
 type Image struct {
 	ID              string            `json:"id"`
 	Parent          string            `json:"parent,omitempty"`
```
Comment (stevvooe) [inline image/image.go]: Please document the individual fields.

File: image/image.go [hunk 6]
```diff
@@ -11,6 +11,18 @@ import (
 
 var validHex = regexp.MustCompile(`^([a-f0-9]{64})$`)
 
+// Image stores the image configuration.
+// ID a unique 64 character identifier of the image
```
Comment (vdemeester) [inline image/image.go]: Should the field be here or inline (in the struct ``Image``).

Comment (brahmaroutu) [inline image/image.go]: I moved the comments into the structure. The doc will be slightly different, comments will be printed with the struct and then the comment about struct.

**Comment by vdemeester** [comment]
@brahmaroutu could you add a period at the end of the doc sentences ? :blush:

**Comment by vdemeester** [comment]
@brahmaroutu could you also add the package ``image`` to ``./hack/make/validate-lint``.

**Comment by LK4D4** [comment]
Cool, LGTM

**Pull request merged**